libevm/cmd/internalise/main.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2024 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+// The internalise command modifies Go files in place, making exported methods
+// internal.
+//
+// Usage:
+//
+//	internalise -file <filepath> <type>.<method> [<type>.<method> [...]]
+//
+// For example, with file foo.go containing declarations:
+//
+//	func (f *Foo) Bar() { ... }
+//
+//	func (Foo) Baz() { ... }
+//
+// running
+//
+//	internalise -file foo.go Foo.Bar Foo.Baz
+//
+// results in
+//
+//	func (f *Foo) bar() { ... }
+//
+//	func (Foo) baz() { ... }
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"os"
+	"strings"
+)
+
+func main() {
+	file := flag.String("file", "", "File to modify")
```

Comment:
What could be nice would be to pass in a package path instead (a bit like mockgen) and let this program find the member to internalize. Not a priority though and we're really not building a tool here, so feel free to ignore too!

Reply:
I considered doing it that way and then had the exact same thought that "we're really not building a tool here".

It's not something we're planning on using any time soon, but [this experimental code modifier](https://github.com/ava-labs/subnet-evm/tree/arr4n/gethclone-stateful-precompiles/x/gethclone) does something similar.